pkg/kubectl/resource/mapper.go

```diff
@@ -64,12 +65,21 @@ func (m *Mapper) InfoForData(data []byte, source string) (*Info, error) {
 	name, _ := mapping.MetadataAccessor.Name(obj)
 	namespace, _ := mapping.MetadataAccessor.Namespace(obj)
 	resourceVersion, _ := mapping.MetadataAccessor.ResourceVersion(obj)
-	return &Info{
-		Mapping:   mapping,
-		Client:    client,
-		Namespace: namespace,
-		Name:      name,
+	var versionedInfo *VersionedInfo
+	versionedObj, _, _, err := api.Scheme.Raw().DecodeToVersionedObject(data)
```

Comment:
I'm pretty sure you don't need to add a VersionedInfo object to `Info`. The api version is already available via `Info.Mapping` which is a [RESTMapping](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/meta/interfaces.go#L128).